### PR TITLE
STY: Enable ruff pytest checks

### PR DESCRIPTION
**Pull request by mroeschke**

### CODE REVIEW

File: pandas/tests/arrays/categorical/test_dtypes.py [hunk 1]
```diff
@@ -73,12 +73,10 @@ def test_set_dtype_new_categories(self):
             (["a", "b", "c"], ["a", "b"], ["a", "b"]),
             (["a", "b", "c"], ["a", "b"], ["b", "a"]),
             (["b", "a", "c"], ["a", "b"], ["a", "b"]),
-            (["b", "a", "c"], ["a", "b"], ["a", "b"]),
             # Introduce NaNs
             (["a", "b", "c"], ["a", "b"], ["a"]),
             (["a", "b", "c"], ["a", "b"], ["b"]),
             (["b", "a", "c"], ["a", "b"], ["a"]),
-            (["b", "a", "c"], ["a", "b"], ["a"]),
```
Comment (lithomas1) [inline pandas/tests/arrays/categorical/test_dtypes.py:81]: Should this be "b"?

File: pandas/tests/arrays/categorical/test_dtypes.py [hunk 2]
```diff
@@ -73,12 +73,10 @@ def test_set_dtype_new_categories(self):
             (["a", "b", "c"], ["a", "b"], ["a", "b"]),
             (["a", "b", "c"], ["a", "b"], ["b", "a"]),
             (["b", "a", "c"], ["a", "b"], ["a", "b"]),
-            (["b", "a", "c"], ["a", "b"], ["a", "b"]),
```
Comment (lithomas1) [inline pandas/tests/arrays/categorical/test_dtypes.py:76]: Should this end with ["b", "a"]?

File: pandas/tests/extension/base/setitem.py [hunk 3]
```diff
@@ -210,7 +210,8 @@ def test_setitem_integer_array(self, data, idx, box_in_series):
                 [0, 1, 2, pd.NA], True, marks=pytest.mark.xfail(reason="GH-31948")
             ),
             (pd.array([0, 1, 2, pd.NA], dtype="Int64"), False),
-            (pd.array([0, 1, 2, pd.NA], dtype="Int64"), False),
+            # TODO: change False to True?
+            (pd.array([0, 1, 2, pd.NA], dtype="Int64"), False),  # noqa: PT014
```
Comment (lithomas1) [inline pandas/tests/extension/base/setitem.py:214]: Hm, this is a dupe right?

File: pandas/tests/extension/base/setitem.py [hunk 4]
```diff
@@ -333,7 +334,7 @@ def test_setitem_loc_iloc_slice(self, data):
 
     def test_setitem_slice_mismatch_length_raises(self, data):
         arr = data[:5]
-        with pytest.raises(ValueError):
+        with tm.external_error_raised(ValueError):
```
Comment (lithomas1) [inline pandas/tests/extension/base/setitem.py:337]: Just to be sure, the error here isn't raised by us?

File: pandas/tests/extension/test_numpy.py [hunk 5]
```diff
@@ -374,19 +374,6 @@ def test_setitem_mask(self, data, mask, box_in_series):
...
-        "idx, box_in_series",
-        [
-            ([0, 1, 2, pd.NA], False),
-            pytest.param([0, 1, 2, pd.NA], True, marks=pytest.mark.xfail),
-            (pd.array([0, 1, 2, pd.NA], dtype="Int64"), False),
-            (pd.array([0, 1, 2, pd.NA], dtype="Int64"), False),
-        ],
-        ids=["list-False", "list-True", "integer-array-False", "integer-array-True"],
-    )
-    def test_setitem_integer_with_missing_raises(self, data, idx, box_in_series):
```
Comment (lithomas1) [inline pandas/tests/extension/test_numpy.py:387]: Do you know why this test got deleted?

File: pandas/tests/extension/base/setitem.py (hunk 3 above)
Comment (mroeschke) [reply pandas/tests/extension/base/setitem.py:214]: Yup but changing it to `True` fails a few tests so adding a noqa for now

File: pandas/tests/extension/base/setitem.py (hunk 4 above)
Comment (mroeschke) [reply pandas/tests/extension/base/setitem.py:337]: I believe not, no

File: pandas/tests/extension/test_numpy.py (hunk 5 above)
Comment (mroeschke) [reply pandas/tests/extension/test_numpy.py:387]: The test definition here is essentially the same as the base test

File: pandas/tests/arrays/categorical/test_dtypes.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/arrays/categorical/test_dtypes.py:76]: Good catch. Fixed

File: pandas/tests/arrays/categorical/test_dtypes.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/arrays/categorical/test_dtypes.py:81]: Good catch. Fixed

File: pandas/tests/frame/test_constructors.py [hunk 6]
```diff
@@ -1555,7 +1555,6 @@ def test_constructor_mixed_type_rows(self):
         "tuples,lists",
         [
             ((), []),
-            ((()), []),
```
Comment (lithomas1) [inline pandas/tests/frame/test_constructors.py:1558]: Hm, was this supposed to contain an inner tuple like
``((),)``

**Review by lithomas1 — COMMENTED** [review]
Left a couple more comments. 

Thanks for cleaning these up by the way!

File: pandas/tests/extension/base/setitem.py (hunk 3 above)
Comment (lithomas1) [reply pandas/tests/extension/base/setitem.py:213]: Can you open an issue as a followup?

(the noqa makes it look like ruff had a false positive)

File: pandas/tests/frame/test_constructors.py (hunk 6 above)
Comment (mroeschke) [reply pandas/tests/frame/test_constructors.py:1558]: Ah possibly yes. That passes here so added it

File: pandas/tests/extension/base/setitem.py (hunk 3 above)
Comment (mroeschke) [reply pandas/tests/extension/base/setitem.py:213]: Sure, https://github.com/pandas-dev/pandas/issues/56727

**Review by WillAyd — APPROVED** [review]
pretty cool - nice work!

**Pull request merged**